lifecycle/CMakeLists.txt

```diff
@@ -23,22 +23,22 @@ add_executable(lifecycle_talker
   src/lifecycle_talker.cpp)
 ament_target_dependencies(lifecycle_talker
   "lifecycle_msgs"
+  "rclcpp"
```

Comment:
The CMakeLists.txt in `rclcpp_lifecycle` [export the dependencies](https://github.com/ros2/rclcpp/blob/b6a803f48ce4c372363596afd8555ccfaa2c40b8/rclcpp_lifecycle/CMakeLists.txt#L159) for `rclcpp`, it seems unnecessary to add `rclcpp` here.

Of course, adding `rclcpp` explicitly is also correct.